x-pack/test/functional/apps/security/rbac_phase1.js

```diff
@@ -0,0 +1,135 @@
+/*
+* Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+* or more contributor license agreements. Licensed under the Elastic License;
+* you may not use this file except in compliance with the Elastic License.
+*/
+
+import expect from 'expect.js';
+import { indexBy } from 'lodash';
+export default function ({ getService, getPageObjects }) {
+
+  const PageObjects = getPageObjects(['security', 'settings', 'monitoring', 'discover', 'common', 'reporting', 'visualize', 'header']);
+  const log = getService('log');
+  const esArchiver = getService('esArchiver');
+  const remote = getService('remote');
+  const kibanaServer = getService('kibanaServer');
+
+  describe('rbac ', async function () {
+    before(async () => {
```

Comment:
nit: generally we use a `before` for setup logic, but we're doing `expect` in here, it feels more like a test.

Reply:
I understand, just wanted to make sure the before method did all the set up before the actual test kicked in. Would consider this as  a part of set up logic.

Reply:
I'll defer to @LeeDr on this one, I didn't see any `expect` calls in the other functional tests, which is what made me mention it here. The one downside is that the failure reasons when the test fail are going to be rather unclear if just the `before` fails as we have no context.

Reply:
@kobelb I asked @rasroh to move the role and user creation code from tests (how she originally wrote it) to the before method because we already have other automated tests creating roles and users and didn't want these setup steps to count as additional passing tests when they would actually be duplicates.  I realize we care more about getting the test coverage than the count of tests, but it just seemed more fair for it to be in the before method.
I think we should still get acceptable failure information if the expects in the before method fail vs having them in a test.  There's debug logging before those expects including showing the list of users.
I asked @rasroh to change one of the expects in the before so it would fail.  Seems like reasonable output;
```
       │ info  Saving page source to: /Users/rashmikulkarni/my_workspace/kibana/x-pack/test/functional/failure_debug/html/security app rbac  _before all_ hook.html
     └- ✖ fail: "security app rbac  "before all" hook"
     │       
     │         Error: expected [ 'rbac_all' ] to sort of equal [ '' ]
     │         + expected - actual
     │       
     │          [
     │         -  "rbac_all"
     │         +  ""
     │          ]
     │         
     │         at Assertion.assert (node_modules/expect.js/index.js:96:13)
     │         at Assertion.eql (node_modules/expect.js/index.js:230:10)
     │         at Context.before (test/functional/apps/security/rbac_phase1.js:54:41)
     │         at <anonymous>
     │       
     │       
       │
       │0 passing (52.1s)
       │1 failing
```